File: backend/app/Models/Address.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Address extends Model
{
    protected $fillable = [
        'user_id', 'name', 'street', 'city', 'state', 
        'postal_code', 'country', 'lat', 'lng', 'is_default'
    ];

    protected $casts = [
        'lat' => 'decimal:8',
        'lng' => 'decimal:8',
        'is_default' => 'boolean',
    ];

    public function user() {
        return $this->belongsTo(User::class);
    }
}

File: flutter_store_ui/lib/models/address.dart
class Address {
  final int id;
  final String name;
  final String street;
  final String city;
  final String state;
  final String postalCode;
  final String country;
  final bool isDefault;
  final double latitude;
  final double longitude;

  Address({
    required this.id,
    required this.name,
    required this.street,
    required this.city,
    required this.state,
    required this.postalCode,
    required this.country,
    required this.isDefault,
    required this.latitude,
    required this.longitude,
  });

  factory Address.fromJson(Map<String, dynamic> json) {
    try {
      return Address(
        id: json['id'] ?? 0,
        name: json['name'] ?? 'Unknown',
        street: json['street'] ?? '',
        city: json['city'] ?? '',
        state: json['state'] ?? '',
        postalCode: json['postal_code'] ?? '',
        country: json['country'] ?? '',
        isDefault: json['is_default'] == 1 || json['is_default'] == true,
        latitude: json['lat'] == null
            ? 0.0
            : double.parse(json['lat'].toString()),
        longitude: json['lng'] == null
            ? 0.0
            : double.parse(json['lng'].toString()),
      );
    } catch (e) {
      print('❌ Error parsing address: $json, Error: $e');
      rethrow;
    }
  }
}
